### ISSUE TITLE
How do I copy files to the directory of build?

### ISSUE THREAD

**Issue by el-leijona** [lukeed/taskr#107]
I have such a directory structure:

```
src/templates/
|_ first.html
|_ second.html
```

and I would like to have all of these html files wind up in one directory side-by-side rather than have the entire tree re-created. However, that code stores the directory structure into `build` folder:

```
export async function resource () {
  await this.source('src/templates/*.html').target('build')
}
```

i.e `build/src/templates/first.html, build/src/templates/second.html`
OS Win10

**Comment by ghost** [comment]
@ljonya This seems like a use case for a `fly-copy` plugin.

``` js
export default function* () {
  yield this.source("src/template/*.html").copy({ ... })
}
```

or perhaps we could accommodate this functionality inside `Fly.proto.target`? How does the resulting code should look like in your opinion?

**Comment by shimarulin** [comment]
@ljonya  It works similarly to Gulp. Which is good, the structure must be retained. Possible solutions were discussed [here](http://stackoverflow.com/questions/21153338/flatten-glob-down-to-one-directory) - possible used the [glob](https://github.com/isaacs/node-glob) library?
@bucaran IMHO enough plugin (maybe `fly-flatten`?)

**Comment by ghost** [comment]
We are using node-glob in `fly-util`, so perhaps adding support for glob options there is the way to go...

`fly-flatten` seems available :smile:

**Comment by shimarulin** [comment]
> fly-flatten seems available :smile: 

Sorry, but i did not find it :smile: Adding support for glob options - it's a good idea

**Comment by ghost** [comment]
Available to claim :grin:

**Comment by TheVoidSeeker** [comment]
I also have this problem, so i tried to create my own filter/plugin for this, but I can't find a way
to get the file name of the current file. Your examples only use the file content and never the file name.
I tried to find the implementation of your 'rename' function, which should be similar, but i couldn't locate it anywhere.

**Comment by lukeed** [comment]
@TheVoidSeeker You may get filenames by `unwrap`ping them.

``` js
this.plugin = function (options) {
  // ...
  this.unwrap(filenames => {
    console.log('These are all my file names: ', filenames)
  })
}
```

**Comment by lukeed** [comment]
@ljonya @bucaran I will make this `flatten` plugin sometime today. Sorry for the delay, I just came aboard :anchor:

**Comment by ghost** [comment]
Ahoy!

**Comment by lukeed** [comment]
:joy:

**Comment by lukeed** [comment]
Closed via #135
